File: C++Stuff/2021F/F1JC.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

#define pb push_back
#define vt vector
#define ll long long

using namespace std;

int solve(){
    int n;
    int ans = 0;
    string poo;
    vt<int> inters;
    cin >> n;
    cin >> poo;

    for (int i; i < n; i++){
        if (poo[i] == '1'){
            inters.pb(i);
        }
    }

    for (int i; i < n; i++){
        int lowest = 100000;
        if (poo[i] == '0'){
            for (int j; j < inters.size(); j++){
                int lowest = abs(inters[j] - i);
                if (int(abs(inters[j] - i)) < lowest){
                    lowest = int(abs(inters[j] - i));
                }
            }
        }
        cout << ans << endl;
        if (poo[i] == '0'){
            ans += lowest;
        }
    }

    return ans;

}


int main(){
    int t, pee;
    cin >> t;
    for (int i; i < t; i++){
        pee = solve();
        cout << "Case #" << i+1 << ": " << pee;
        cout << endl;
    }

    return 0;
}

File: C++Stuff/2021F/F1JC.py
t = int(input())



def solve():
    ans = 0
    n = int(input())
    poo = str(input())

    inters = []

    for i in range(n):
        if poo[i] == '1':
            inters.append(i)

    for i in range(n):
        lowest = 100000000
        if poo[i] == '0':
            inters.append(i)
            inters.sort()
            for j in range(-1,2,2):
                try:
                    if abs(inters[inters.index(i)] - inters[inters.index(i)-j]) < lowest:
                        lowest = abs(inters[inters.index(i)] - inters[inters.index(i)-j])
                except IndexError:
                    continue
            inters.remove(i)

            ans += lowest



    return ans



for i in range(t):
    print(f"Case #{i+1}: {solve()}" )








